bokeh/models/sources.py

```diff
@@ -256,6 +256,14 @@ def _check_column_lengths(self):
         if len(lengths) > 1:
             return str(self)
 
+
+class GeoJSONDataSource(DataSource):
+    geojson = JSON(help="""
```

Comment:
Would be fairly simple to add a `class GeoJSON(JSON)` property that does some validation, makes sure that there are only GeometryCollection and FeatureCollection at top level, for instance.